### PR TITLE
Adds guidelines for graded browser support.

### PR DESCRIPTION
**Pull request by sky-jack**

### CODE REVIEW

File: practices/graded-browser-support.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+## Conclusion
+
+Graded Browser Support provides an inclusive definition of support and a framework for taming the ever-expanding world of browsers and frontend technologies.
+
+Tim Berners-Lee, inventor of the World Wide Web and director of the W3C, has said it best:
+
+> > "Anyone who slaps a 'this page is best viewed with Browser X' label on a Web page appears to be yearning for the bad old days, before the Web, when you had very little chance of reading a document written on another computer, another word processor, or another network."
+
+
+- Reproduced from guidelines by [http://nate.koechley.com/](Nate Koechely)
```
Comment (joseluisbolos) [inline practices/graded-browser-support.md]: Are we citing the source properly? I can't seem to find the original post, but his site has a CC BY 2.5 license, which states:

> Attribution — You must give appropriate credit, provide a link to the license, and indicate if changes were made.

Any idea where the source is? @dotcode ?

Comment (hollsk) [inline practices/graded-browser-support.md]: There's one small (and by "small" i mean "enormous") hitch when it comes to the source - we reproduced this from the Yahoo! YUI guidelines: https://github.com/yui/yui3/wiki/Graded-Browser-Support

It's also open source but it gives me a bad feeling to reproduce it verbatim, even if it's correctly attributed. Perhaps we should keep the table and anything that isn't included in the YUI source, and link out to their source instead.

**Comment by dotcode** [comment]
@joseluisbolos just emailing Nate to find out :)

**Comment by dotcode** [comment]
@hollsk yeah … we're copying it _almost_ verbatim, the bits we've altered refer to QA support for that strategy. Happy to do what you suggest though, it makes sense (the edits we've made are trivial). Can always fork it if necessary in future.

File: practices/graded-browser-support.md (hunk 1 above)
Comment (sky-jack) [inline practices/graded-browser-support.md]: Agreed, keeping the table and linking out to source seems best.

**Comment by hollsk** [comment]
We had a bit of chat about it in Slack, and Jose and I thought that one approach might be to copy what YUI did by having the table on one page, and some more supporting blurb on another. That would give us the best of both worlds, I think.

File: practices/graded-browser-support.md [hunk 2]
```diff
@@ -12,88 +19,34 @@
...
 
-Graded Browser Support provides an inclusive definition of support and a framework for taming the ever-expanding world of browsers and frontend technologies.
+```html
+<link href="style.css" rel="stylesheet"  media="only screen and (-webkit-min-device-pixel-ratio:0), (min-color-index:0), (-ms-high-contrast: none)" />
 
-Tim Berners-Lee, inventor of the World Wide Web and director of the W3C, has said it best:
+```
 
-> > "Anyone who slaps a 'this page is best viewed with Browser X' label on a Web page appears to be yearning for the bad old days, before the Web, when you had very little chance of reading a document written on another computer, another word processor, or another network."
+We initialises JavaScript only if the browser supports a number of modern:
```
Comment (hollsk) [inline practices/graded-browser-support.md]: I think you lost your train of thought while you were editing this line 😁

Comment (sky-jack) [inline practices/graded-browser-support.md]: initaliseseses

**Comment by hollsk** [comment]
Do it 👍

File: practices/graded-browser-support.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Graded Browser Support
+
+
+## Our criteria for browser support
+
+
+We followe the guidelines for graded browser support outlined by Nate Koechley at Yahoo:
```
Comment (benjclark) [inline practices/graded-browser-support.md]: follow typo

File: practices/graded-browser-support.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+[https://github.com/yui/yui3/wiki/Graded-Browser-Support](Yahoo Grade Browser Support guide)
```
Comment (benjclark) [inline practices/graded-browser-support.md]: if i preview these changes this link doesn't work, says page not found

Comment (benjclark) [inline practices/graded-browser-support.md]: This applies to all the links actually. I think it's because the format should be [name](url)

Comment (davidpauljunior) [inline practices/graded-browser-support.md]: `[Yahoo Grade Browser Support guide](https://github.com/yui/yui3/wiki/Graded-Browser-Support)`

Comment (sky-jack) [inline practices/graded-browser-support.md]: ah yes. Silly me.

**Comment by sky-jack** [comment]
Fixed typos.

**Comment by benjclark** [comment]
👍

File: practices/graded-browser-support.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+| Webkit        | Android 4.&#8224; |  <3           |
+
+
+
+
+
+## How we implement graded browser support
+
+
+There are currently two methods by which implement graded support for browsers using principles of progressive enhancement.
```
Comment (umaar) [inline practices/graded-browser-support.md]: From:
> There are currently two methods by which implement

To:
> There are currently two methods by which _we_ implement

**Comment by sky-jack** [comment]
All sounds good. Should have re-read more carefully!

**Comment by joseluisbolos** [comment]
Those values on the table don't match what we currently have on the code. Seems like someone may have changed them (incorrectly) at some point.

**Comment by sky-jack** [comment]
What should change?

**Comment by joseluisbolos** [comment]
Gosh, I missed this thread. This is the current list of C-Grade browsers in our legacy platform (which we've been using as a _single source of truth_ so far):

* Internet Explorer <9
* Safari up to 4.0.x on Windows and OS X
* Safari 2 (any) and Safari 3 (iPhone 1)
* Unknown (User-Agent matches '.* AppleWebKit.*Version/[0-2]\..*')
* Safari 1
* Unknown (User-Agent matches '.*Version/2\.2 Firefox/525\.13')
* **Firefox 0.x and 1**
* **Opera up to 9.4**
* Nokia browser for Symbian 7.3
* Android browser up to 2.x
* Old versions of 'Mozilla compatible' browsers
* Netscape

In *bold* the ones that don't match what's on the table. As you can see, there are others that are not on the table at all, I'm not sure what we want to do and if we want to put them there.

**Comment by hollsk** [comment]
I'm merging this _right now_ as it's a blocker for some other updates. I've opened issue #117 to correct the C-grade values, which should get things moving again :-)

**Pull request merged**